src/microservices/Microservices.IsIdentifiable/Rules/WhiteListRule.cs

```diff
@@ -50,35 +48,25 @@ public class WhiteListRule : ICustomRule
         public string IfPattern
         {
             get => _ifPattern?.ToString();
-            set => _ifPattern = value == null ? null : new Regex(value, RegexOptions.IgnoreCase | RegexOptions.Compiled);
+            set => _ifPattern = value == null ? null : new Regex(value, (IfPatternIsCaseSensitive ? 0 : RegexOptions.IgnoreCase) | RegexOptions.Compiled);
```

Comment:
What is the difference between WhiteListRule and IsIdentifiableRule? do we need both?

Reply:
WhiteListRules have two patterns, one to match the whole string and one to match the substring part which previously matched a rule.

Reply:
Ah ok cool, can we cut down on code duplication by inheriting one from the other? (just an idea)